src/tooling/grain_doc.md

```diff
@@ -0,0 +1,148 @@
+
+The `@returns` attribute provides information for the return value of the function being documented.
+
+This attribute requires a description of the return value. In the Grain standard library, we don't use the `@returns` attribute if the function returns `Void`.
+
+When generating docs, the type of the return value will be detected by the compiler and added to the row in the "Returns" table.
+
+### @example
+
+```gr
+/**
+ * @example code
+ */
+```
+
+The `@example` attribute provides snippets showing how the documented code is used. Currently, only single line examples are allowed.
+
+When generating docs, the code snippet will be wrapped in a code block tagged for the `grain` language and added to an "Examples" section.
+
+### @since
+
+```gr
+/**
+ * @since version
+ */
+```
+
+The `@since` attribute provides the version in which the documented code was added.
+
+This attribute requires a [Semantic Version](https://semver.org/), optionally prefixed with `v`. For example,`v0.2.0` or `0.2.0` are treated the same.
+
+When generating docs, a `<details>` element titled "Added in X.X.X" will be added above the type signature. When this attribute is specified, you'll be required to specify the `--current-version=X.X.X` flag to the `grain doc` command. This allows you to specify newer versions in your source code, and the generated docs will show "Added in next", instead of the version.
+
+### @history
+
+```gr
+/**
+ * @history version: description
+ */
+```
+
+The `@history` attribute provides details on significant changes to the documented code.
+
+This attribute requires a [Semantic Version](https://semver.org/) before the colon (`:`) and a description of the change after the colon.
+
+When generating docs, adds a history table to the `<details>` element created by the `@since` attribute if both were used; otherwise, a new `<details>` element titled  "History" will be inserted above the type signature. When using this attribute, you'll be required to specify the `--current-version=X.X.X` flag to the `grain doc` command. This allows you to specify newer versions in your source code, and the generated docs will show "next" instead of the version.
+
+### @deprecated
+
+```gr
+/***
+ * @deprecated description
+ */
+```
+
+The `@deprecated` attribute provides a description or reason for the deprecation of the documented code.
+
+This attribute requires a description of the deprecation and will mark the export as deprecated when used.
+
+When generating docs, adds a blockquote containing `**Deprecated:**` and the description immediately below the title.
+
+### @module
+
+```gr
+/***
+ * @module Array: Utilities for working with arrays.
+ *
+ * @example import Array from "array"
+ *
+ * @since v0.2.0
+ * @history v0.1.0: Originally named `arrays`
+ * @history v0.2.0: Renamed to `array`
+ */
+```
+
+The `@module` attribute allows you to add header documenting to a module. A module docblock can contain `@example`, `@since`, and `@history` attributes.
```

Comment:
```suggestion
The `@module` attribute provides top-level information about a file, including the module title and a description.

This special attribute is used within a docblock comment that is not attached to a type declaration or exported value. It can only be specified once per file and should exist near the top.

The `@module` docblock comment can also contain `@example`, `@since`, and `@history` attributes that will add their corresponding output to the top of the documentation.

When generating docs, adds the description at the top of the documentation, and a [Front Matter](https://jekyllrb.com/docs/front-matter/) section containing the title.
```